SPHINXsys/src/shared/body_relations/base_body_relation.h

```diff
@@ -143,18 +143,19 @@ namespace SPH
 	class BaseContactRelation : public SPHRelation
 	{
 	protected:
+		UniquePtrsKeeper<ParticleConfiguration> configuration_ptrs_keeper_;
 		virtual void resetNeighborhoodCurrentSize();
 
 	public:
 		RealBodyVector contact_bodies_;
-		ContactParticleConfiguration contact_configuration_; /**< Configurations for particle interaction between bodies. */
+		StdVec<ParticleConfiguration *> contact_configuration_; /**< Configurations for particle interaction between bodies. */
```

Comment:
There is no visible benefits to this. Value semantic is preferable, it just needs a few methods around the original container to enforce security and proper usage.

Reply:
Let me reformulate, what I meant is that I don't see the reason to replace `ContactParticleConfiguration` by `UniquePtrsKeeper<ParticleConfiguration>`, while making the member variable protected/private with proper set of getter would do the same.